### PR TITLE
Package bnfgen.2.0

### PR DESCRIPTION
**Pull request by dmbaturin**
### `bnfgen.2.0`
Random text generator that takes context-free grammars from BNF files
BNFGen generates random texts based on user-defined context-free grammars
specified in a BNF-like syntax.
You can specify "weight" for rules with alternation to change the probabilities,
for example, in `<foo> ::= 10 <foo> "foo" | "foo";` the recursive option will be
ten times more likely to be taken.

---
* Homepage: https://github.com/dmbaturin/bnfgen
* Source repo: git+https://github.com/dmbaturin/bnfgen
* Bug tracker: https://github.com/dmbaturin/bnfgen/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

**Comment by dmbaturin** [comment]
This is a new release and a rename of https://opam.ocaml.org/packages/cfgen/

I've mostly been using it for my own needs, but then I found that there are two more projects with that name, for different purposes (one not related to formal languages at all, the other extracts grammars from a corpus rather than takes an explicit definition), so I tried to make the name more descriptive.

I couldn't find if there's a procedure for renaming opam packages of whether it ever happened before. If any steps are needed to properly rename the package, please let me know.

**Comment by avsm** [comment]
Thanks! It depends what you prefer -- if there are no other packages depending on it, then you can do a repository rename. Alternatively just put a prominent message in the metadata of the old package that it haas been deprecated and to direct users to the new name.

**Comment by dmbaturin** [comment]
@avsm That's the reason I decided to go through with the rename really—nothing _can_ depend on it because it only provides executables, not modules (for better or worse, maybe something to consider for a future update, along with ABNF support).

I'll make a separate pull request to add a deprecation warning to the original cfgen then. jbuilder seems to provide a good example of how it's done.

File: packages/bnfgen/bnfgen.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+license: "MIT"
+homepage: "https://github.com/dmbaturin/bnfgen"
+bug-reports: "https://github.com/dmbaturin/bnfgen/issues"
+dev-repo: "git+https://github.com/dmbaturin/bnfgen"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name]
+]
+depends: [
+  "ocamlfind" {build}
```
Comment (mseri) [inline packages/bnfgen/bnfgen.2.0/opam]: Is ocamlfind really still used in the build?

Comment (dmbaturin) [reply packages/bnfgen/bnfgen.2.0/opam]: Good point. I've updated the PR.

**Comment by mseri** [comment]
Have you modified the tarball?
```
bnfgen.2.0:
            https://github.com/dmbaturin/bnfgen/archive/2.0.tar.gz (Bad checksum, expected md5=89ae8d4e9db009b8b54ebcc9df18715d)
```

**Comment by dmbaturin** [comment]
In this case, I have to admit, yes. I've found a stupid mistake and thought it was safe to sneakily overwrite the tag until the PR is merged and get away with it. Sorry for making this PR so much trouble.

File: packages/bnfgen/bnfgen.2.0/opam [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+homepage: "https://github.com/dmbaturin/bnfgen"
+bug-reports: "https://github.com/dmbaturin/bnfgen/issues"
+dev-repo: "git+https://github.com/dmbaturin/bnfgen"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name]
+]
+depends: [
+  "menhir" {build}
+  "dune" {build & >= "1.4.0"}
```
Comment (mseri) [inline packages/bnfgen/bnfgen.2.0/opam]:
```suggestion
  "dune" {>= "1.4.0"}
```

**Comment by mseri** [comment]
If you update the opam file and use the correct checksum for the tarball, this could be merged

File: packages/bnfgen/bnfgen.2.0/opam (hunk 2 above)
Comment (dmbaturin) [reply packages/bnfgen/bnfgen.2.0/opam]: Fixed, please check.

**Comment by mseri** [comment]
Thanks. The CI is a bit busy right now. I'll wait for Travis to do its job before merging.

**Pull request merged**